controller/volume_controller.go

```diff
@@ -1317,6 +1254,15 @@ func (vc *VolumeController) replenishReplicas(v *longhorn.Volume, e *longhorn.En
 		return nil
 	}
 
+	if hardNodeAffinity != "" && getRebuildingReplicaCount(e) == 0 {
+		r, err := vc.createReplica(v, e, rs, hardNodeAffinity)
+		if err != nil {
+			return err
+		}
+		rs[r.Name] = r
+		return nil
```

Comment:
hmm? Should we skip all the logic below? This should be automatically taken care of.

Reply:
Do you mean that we can skip this logic?
```
if hardNodeAffinity != "" && getRebuildingReplicaCount(e) == 0 {
 		r, err := vc.createReplica(v, e, rs, hardNodeAffinity)
 		if err != nil {
 			return err
 		}
 		rs[r.Name] = r
 		return nil
```

Reply:
I meant we shouldn't need to `return nil` here.

Reply:
Yeah, we don't need to return nil. Thank you